kensu/mlflow/pyfunc/__init__.py

```diff
@@ -0,0 +1,148 @@
+from mlflow.pyfunc import *
+from kensu.client import DataSourcePK, DataSource, Schema, SchemaPK, FieldDef
+
+def wrap_log_model(method):
+    def wrapper(*args, **kwargs):
+        kensu = KensuProvider().instance()
+        result = method(*args, **kwargs)
+
+        if 'python_model' in kwargs:
+            model_list = []
+            python_model = kwargs['python_model']
+            python_model_dict = python_model.__dict__
+            for attribute in python_model_dict:
+                potential_model = python_model_dict[attribute]
+                if ('kensu.sklearn') in str(potential_model.__class__):
+                    model_list.append(potential_model)
+
+            run_id = mlflow.tracking.fluent._get_or_start_run().info.run_id
+            model_log_location = 'runs:/' + run_id + '/' + args[0]
+            model_log_location = mlflow.get_artifact_uri(args[0])
+
+            i = 0
+            for model in model_list:
+                i += 1
+                if ('kensu.sklearn') in str(potential_model.__class__):
+                    fmt = 'Sklearn Model'
+
+                    orig_ds = eventually_report_in_mem(kensu.extractors.extract_data_source(model.attr[0], kensu.default_physical_location_ref,
+                                                                 logical_naming=kensu.logical_naming,format=fmt))
+                    orig_sc = eventually_report_in_mem(kensu.extractors.extract_schema(orig_ds, model.attr[0]))
+
+                    location = model_log_location +'/part-'+str(i)
+
+                    result_ds = kensu.extractors.extract_data_source(model.attr[0], kensu.default_physical_location_ref, location = location,
+                                                                   logical_naming=kensu.logical_naming,format=fmt)._report()
+                    result_sc = kensu.extractors.extract_schema(result_ds, model.attr[0])._report()
+
+                    kensu.register_alias(ref_guid=orig_sc.to_guid(), alias_guid=result_sc.to_guid())
+                    kensu.real_schema_df[result_sc.to_guid()] = model
+                    kensu.model[result_sc.to_guid()] = model.attr
+
+                    #TODO create mlflow extractor
+                    from mlflow.pyfunc import PythonModel
+                    if python_model.__class__ in PythonModel.__subclasses__():
+                        ds_pk = DataSourcePK(location=model_log_location,
+                                             physical_location_ref=kensu.default_physical_location_ref)
+                        name = args[0]
+                        fields = [
+                            FieldDef(name="intercept_", field_type="float", nullable=False),
+                            FieldDef(name="coef_", field_type="array<float>", nullable=False)
+                        ]
+                        logged_ds = DataSource(name=name,categories=['logical::'+name], format='MLFlow', pk=ds_pk)._report()
+                        sc_pk = SchemaPK(logged_ds.to_ref(),
```

Comment:
we use low level constructs over and over again. we could re-use helpers and have less code and easier to read...